### PR TITLE
Fourier operator base

### PR DESCRIPTION
**Pull request by paquiteau**
This PR creates a BaseFourierClass, which represent the minimal interface every Fourier Operator of pysap-mri should have: 
- an `adj_op` method
- an `op` method
- an `uses_sense` attribute/property 
- `n_coils` and `shape` attributes

This Make all the Fourier operator of pysap-mri inherit this class. 

Bonus: This make the `SingleChannelReconstructor` able to reconstruct using SENSE, if the fourier operator can do it.

### CODE REVIEW

File: mri/operators/fourier/base.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        Returns
+        -------
+        results: np.ndarray
+            adjoint operator transform.
+        """
+        raise NotImplementedError("'adj_op' is an abstract method.")
+
+    @property
+    def uses_sense(self):
+        """Return True if the operator uses sensitivity maps."""
```
[the remark this answers is not in the document]
Comment (chaithyagr) [reply mri/operators/fourier/base.py:69]: We can add reference here also

File: mri/operators/fourier/non_cartesian.py [hunk 2]
```diff
@@ -384,8 +384,15 @@ def adj_op(self, coeffs, *args):
         else:
             return self.impl.adj_op(coeffs, *args)
 
+    @property
+    def uses_sense(self):
+        """Return True if the Fourier Operator uses the SENSE method."""
+        try:
+            return self.impl.uses_sense
```
Comment (chaithyagr) [inline mri/operators/fourier/non_cartesian.py:391]: How is this initialized? I dont see it anywhere in this PR.

**Review by chaithyagr — CHANGES_REQUESTED** [review]
I have some minor concerns and that you have 2 PRs issues in same branch. We can either revert these changes, or just add the details of other PR here and we can just merge it here.

File: mri/operators/fourier/non_cartesian.py (hunk 2 above)
Comment (paquiteau) [reply mri/operators/fourier/non_cartesian.py:391]: It's not initialized. By default its False for the base operator.

File: mri/operators/fourier/base.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (paquiteau) [reply mri/operators/fourier/base.py:69]: Do you have any good reference available ?

**Review by paquiteau — COMMENTED** [review]
Lets also consider the dtype modification  here (which are minor), I will drop #153 .

**Review by Daval-G — CHANGES_REQUESTED** [review]
One quick comment, otherwise LGTM

**Review by Daval-G — APPROVED** [review]

**Comment by chaithyagr** [comment]
LGTM..

**Pull request merged**